woocommerce/woocommerce
# Wrangle MarketingRecommendations as RemoteSpecsEngine

### Changes proposed in this Pull Request:

Follow-up to https://github.com/woocommerce/woocommerce/pull/44775

- Add RemoteSpecsEngine class to MarketingRecommendations
- Add return default when empty specs
- Fix potential recursion issue
- Add error handling
- Add tests



### How to test the changes in this Pull Request:

#### Test expected behaviour in Marketing page

1. Enable marketplace suggestions in WooCommerce Settings > Advanced > Woo.com: `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com`
2. Check that there isn't a transient called `woocommerce_admin_marketing_recommendations_specs`. If there is, delete it. 
3. If there is a transient called `wc_marketing_recommended_plugins`, delete it.
4. Go to Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`
5. We should see marketing channel recommendations. The recommendation data corresponds to the data retrieved from `https://woocommerce.com/wp-json/wccom/marketing-tab/1.3/recommendations.json`.
6. Check that the data is stored in a transient called `woocommerce_admin_marketing_recommendations_specs`. 
7. Check that there is no transient called `wc_marketing_recommended_plugins`.

#### Test artificial error handling

1. Ensure your environment's PHP version is `> 8.1`
1. In `wp-config.php`, ensure `WP_ENVIRONMENT_TYPE` is set to `local` or `development`
2. In your local, add the code `in_array( '', '', true );` into [`Init.php`](https://github.com/woocommerce/woocommerce/blob/245c23935d33b2ad206e098ab8b02b7a4427ae22/plugins/woocommerce/src/Admin/Features/MarketingRecommendations/Init.php#L157) to simulate errors in PHP 8+.
4. Go to Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`
5. Go to `WooCommerce > Status > Logs` and look for logs named `remotespecsengine-errors`
6. In the logs, ensure there are errors that came from the following classes:

```
Automattic\\WooCommerce\\Admin\\Features\\MarketingRecommendations\\Init
```



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
